syllabus/3-Blockchain/2-P2P_Networks_State_Machines-slides.md

```diff
@@ -101,43 +122,147 @@ This led to Metallica filing a lawsuit against Napster for copyright infringemen
 Napster had to comply by banning hundreds of thousands of users from their platform who were sharing Metallica's music.
 This was a turning point in digital copyright law and played a significant role in Napster's eventual shutdown in 2001.
 
----
+---v
 
 ## Napster Setup
 
 <img style="width: 600px" src="./img/napster_1.svg" />
 
----
+---v
 
 ## Napster Setup
 
 <img style="width: 600px" src="./img/napster_2.svg" />
 
----
+---v
 
 ## Napster Setup
 
 <img style="width: 600px" src="./img/napster_3.svg" />
 
----
+---v
 
 ## Napster Setup
 
 <img style="width: 600px" src="./img/napster_4.svg" />
 
 ---
 
-## Gnutella(Limewire)
+## Limewire
+
+Free Software p2p file sharing **client**
+
+<!-- .element: class="fragment" data-fragment-index="2" -->
+
+In its history, it has mainly used two protocols
+
+<!-- .element: class="fragment" data-fragment-index="3" -->
+
+---v
 
-- Each node serves as both a client and a server no central server <!-- .element: class="fragment" data-fragment-index="2" -->
-- Query all connected nodes for files <!-- .element: class="fragment" data-fragment-index="3" -->
-- Gain peer connections to the network via Bootnodes <!-- .element: class="fragment" data-fragment-index="4" -->
-- Ordered to shutdown in 2010 by United States Court <!-- .element: class="fragment" data-fragment-index="5" -->
+## Gnutella
+
+- Each node serves as both a client and a server <!-- .element: class="fragment" data-fragment-index="2" -->
+- Query recursively all connected nodes for files <!-- .element: class="fragment" data-fragment-index="3" -->
+- It does not just flood the network, it does it intelligently using concepts such as:
+  <!-- .element: class="fragment" data-fragment-index="4" -->
+  - Infection Factor
+  <!-- .element: class="fragment" data-fragment-index="5" -->
+  - TTL (Time To Live)
+  <!-- .element: class="fragment" data-fragment-index="5" -->
+
+<img style="width: 400px" src="./img/gnutella.gif" />
+<!-- .element: class="fragment" data-fragment-index="5" -->
 
 Notes:
 
 - Check local filestore for file and if it is not available, forward the request to all connected peers.
-- Gnutella generates a significant amount of network traffic by flooding the network with requests.
+
+- Gnutella generates a significant amount of network traffic
+
+source: https://nakamoto.com/gnutella/
```

Comment:
Yeah, make sure they know this is a good link worth looking at. It is a source for that direct claim about flooding, but it also has a lot more to offer.